### PR TITLE
feat: added data table for transactions of saving account

### PR DESCRIPTION
**Pull request by PC-11-00**
## Description
Now we can add the data table for transactions entity type of saving account and also we can edit,delete and create the datatable.

## Related issues and discussion
#1726 

## Screenshots, if any

https://github.com/openMF/web-app/assets/76156941/f1ac585c-27a0-40fd-a217-41870debf6d8

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

**Comment by bharathcgowda** [comment]
@ramvr1256 @luckyman20   could you please help in reviewing this PR?

File: src/app/savings/common-resolvers/transaction-datatable.resolver.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// Angular Imports
+import { Injectable } from '@angular/core';
+import { Resolve, ActivatedRouteSnapshot } from '@angular/router';
+
+//rxjs Imports
```
Comment (luckyman20) [inline src/app/savings/common-resolvers/transaction-datatable.resolver.ts:5]:
```suggestion
// rxjs Imports 
```

File: src/app/savings/common-resolvers/transaction-datatables.resolver.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// Angular Imports
+import { Injectable } from '@angular/core';
+import { Resolve, ActivatedRouteSnapshot } from '@angular/router';
+
+//rxjs Imports
```
Comment (luckyman20) [inline src/app/savings/common-resolvers/transaction-datatables.resolver.ts:5]:
```suggestion
// rxjs Imports 
```

File: src/app/savings/common-resolvers/transaction-datatables.resolver.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  /**
+   * 
+   * @param savingsService 
+   */
+  constructor(private savingsService:SavingsService){}
+  /**
+   * 
+   * @param route 
+   * @returns {Observable<any>}
```
Comment (luckyman20) [inline src/app/savings/common-resolvers/transaction-datatables.resolver.ts:22]:
```suggestion
   * @returns {Observable<boolean>}
```

File: src/app/savings/savings-account-view/transactions/view-transaction/datatable-transaction-tab/datatable-transaction-tab.component.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import { ActivatedRoute } from '@angular/router';
+
+@Component({
+  selector: 'mifosx-datatable-transaction-tab',
+  templateUrl: './datatable-transaction-tab.component.html',
+  styleUrls: ['./datatable-transaction-tab.component.scss']
+})
+
+export class DatatableTransactionTabComponent implements OnInit {
+  entityId: string;
```
Comment (luckyman20) [inline src/app/savings/savings-account-view/transactions/view-transaction/datatable-transaction-tab/datatable-transaction-tab.component.ts:11]: Need comment for this variable as well.

File: src/app/savings/savings-account-view/transactions/view-transaction/datatable-transaction-tab/datatable-transaction-tab.component.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  multiRowDatatableFlag: boolean;
+  constructor(private route: ActivatedRoute) {
+    this.entityId = this.route.parent.parent.snapshot.paramMap.get('id');
+    this.route.data.subscribe((data: { transactionDatatable:any }) => {
+      this.entityDatatable = data.transactionDatatable;
+      this.multiRowDatatableFlag = this.entityDatatable.columnHeaders[0].columnName === 'id' ? true : false;
+    });
+   }
+
+  ngOnInit(): void {
```
Comment (luckyman20) [inline src/app/savings/savings-account-view/transactions/view-transaction/datatable-transaction-tab/datatable-transaction-tab.component.ts:24]: Is it required if it is empty?

File: src/app/savings/savings-routing.module.ts [hunk 6]
```diff
@@ -141,8 +144,24 @@ const routes: Routes = [
...
+              savingsAccountTransaction: SavingsAccountTransactionResolver,
+              transactionDatatables: TransactionDatatablesResolver
+            },
+            children: [
+              {
+                path: 'datatables',
+                children:[{
+                  path:':datatableName',
+                  component:DatatableTransactionTabComponent,
+                  data: {title : extract('View Data table'),routeParamBreadcrumb:'datatableName'},
```
Comment (luckyman20) [inline src/app/savings/savings-routing.module.ts:156]:
```suggestion
                  data: {title : extract('View Data table'), routeParamBreadcrumb:'datatableName'},
```

File: src/app/savings/savings-routing.module.ts [hunk 7]
```diff
@@ -141,8 +144,24 @@ const routes: Routes = [
...
-            }
+              savingsAccountTransaction: SavingsAccountTransactionResolver,
+              transactionDatatables: TransactionDatatablesResolver
+            },
+            children: [
+              {
+                path: 'datatables',
+                children:[{
+                  path:':datatableName',
+                  component:DatatableTransactionTabComponent,
```
Comment (luckyman20) [inline src/app/savings/savings-routing.module.ts:155]:
```suggestion
                  component: DatatableTransactionTabComponent,
```

File: src/app/savings/savings-routing.module.ts [hunk 8]
```diff
@@ -141,8 +144,24 @@ const routes: Routes = [
...
-              savingsAccountTransaction: SavingsAccountTransactionResolver
-            }
+              savingsAccountTransaction: SavingsAccountTransactionResolver,
+              transactionDatatables: TransactionDatatablesResolver
+            },
+            children: [
+              {
+                path: 'datatables',
+                children:[{
+                  path:':datatableName',
```
Comment (luckyman20) [inline src/app/savings/savings-routing.module.ts:154]:
```suggestion
                  path: ':datatableName',
```

File: src/app/system/manage-data-tables/app-table-data.ts [hunk 9]
```diff
@@ -6,6 +6,7 @@ export const appTableData: { displayValue: string, value: string }[] = [
     { displayValue: 'Loan Account', value: 'm_loan' },
     { displayValue: 'Saving Account', value: 'm_savings_account' },
     { displayValue: 'Loan Product', value: 'm_product_loan' },
+    {displayValue: 'Saving Account Transaction',value: 'm_savings_account_transaction'},
```
Comment (luckyman20) [inline src/app/system/manage-data-tables/app-table-data.ts:9]:
```suggestion
    { displayValue: 'Saving Account Transaction', value: 'm_savings_account_transaction' },
```

File: src/app/savings/common-resolvers/transaction-datatables.resolver.ts [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+
+// Custom Service
+import { SavingsService } from '../savings.service';
+
+@Injectable()
+export class TransactionDatatablesResolver implements Resolve<Object> {
+
+  /**
+   * 
+   * @param savingsService
```
Comment (luckyman20) [inline src/app/savings/common-resolvers/transaction-datatables.resolver.ts:16]:
```suggestion
   * @param savingsService Savings Service.
```

**Review by luckyman20 — CHANGES_REQUESTED** [review]
Please keep an eye on how you comment. The logic seems fine.

@bharathcgowda I hope this is tested.

Tests are failing as well, please see to that.

**Comment by PC-11-00** [comment]
@luckyman20 which tests are failing?

File: src/app/savings/savings-account-view/transactions/view-transaction/datatable-transaction-tab/datatable-transaction-tab.component.ts (hunk 5 above)
Comment (PC-11-00) [reply src/app/savings/savings-account-view/transactions/view-transaction/datatable-transaction-tab/datatable-transaction-tab.component.ts:24]: No, it is not required I can remove it, actually when we generate a component in angular this comes automatically.

File: src/app/savings/common-resolvers/transaction-datatables.resolver.ts (hunk 3 above)
Comment (PC-11-00) [reply src/app/savings/common-resolvers/transaction-datatables.resolver.ts:22]: Actually below should be any

Comment (luckyman20) [reply src/app/savings/common-resolvers/transaction-datatables.resolver.ts:22]: Can you please elaborate?

**Comment by luckyman20** [comment]
> @luckyman20 which tests are failing?

I can see that task named 'Publish Image in Docker Hub / build (pull_request) has failed.'

**Review by luckyman20 — APPROVED** [review]
LGTM!

**Comment by PC-11-00** [comment]
@bharathcgowda please merge it now.

**Comment by bharathcgowda** [comment]
Thank you @luckyman20 for reviewing this properly and thank you @PC-11-00  for fixing all the comments.

**Pull request merged**